src/cljs/athens/views/blocks/textarea_keydown.cljs

```diff
@@ -415,14 +416,34 @@
 
       ;; Else: navigate across blocks
       ;; FIX: always navigates up or down for header because get-caret-position for some reason returns the wrong value for top
+
+      ;; going LEFT at **0th index** should always go to **last index** of block **above**
+      ;; last index is special - always go to last index when going up or down
+
+      ;; NOTE: Using 99999999999 is a hack, if the previous block has less character than mentioned then the default
+      ;;       caret position will be last position. Otherwise, we would have to calculate the no. of characters in the
+      ;;       block we are moving to, this calculation would be done on client side and, I am not sure if the calculation
+      ;;       would be correct because between calculation on client side and block data on server can change.
+
+      (or (and left? start?)
+          (and up? end?))         (do (.. e preventDefault)
+                                      (dispatch [:up uid 99999999999]))
```

Comment:
It would be better to use sentinel `:end` instead of magic number here, and in handler for both `:up` & `:down` calculate what `:end` means.

Reply:
Did you read his comments @neotyk ?

Reply:
Alex to calculate the end we would have to look at the end of block to which caret is moving. This calculation being done on client side could be wrong

Reply:
For example 

Current block text: hello world
Next block text   : Hello world this

To move from current block to next block, we would have to calculate :end position which would be position after "this". 

But it could be possible that between the client calculating the position and sending it to server some other client added more thus making 

Next block text: Hello world this is sid

Reply:
> Did you read his comments @neotyk ?

![Yes](https://media.giphy.com/media/Ma1gjwuWUco5Ze6IIr/giphy.gif)

Reply:
I do get it that it can change.
And that we can feed `setCursorPostion` too big value, it's ok.
What I'd prefer to see here is meaningful semantics, this magic number is accidental complexity of implementation,
we really mean here stick to `:end` of block.
Yes we'll resolve it to a big number, please let's do it close to implementation that requires this accidental complexity.